openLRSng/defines.h

```diff
@@ -0,0 +1,298 @@
+#ifndef _DEFINES_H_
```

Comment:
What is the motivation for centralising all of these defines, instead of keeping them where they are used?

Reply:
The idea was for clarity and to keep things centralized and easy to find, but I admit I might have gotten carried away with some if it.

Ideally, if we were working with a more c/c++ traditional source tree setup, we'd have things separated by header+source file, with defines, prototypes, & typedefs all stored within the header, and the source file nice & clean. However, like I mentioned up above, since we're dealing with a bunch of headers, I figured it might be easier to just dump all of the macros & constants in one file so that the rest of the header files wouldn't be cluttered up with a bunch of preprocessor stuff up top.

In theory, we could probably switch everything over to the more traditional way of separating the code into 'cpp source + header', although it might cause some issues with Arduino if we don't do it right. If you're into that idea, I'd be willing to try to set that up instead...

Reply:
I think it is totally right for compilation units to have private defines a the top. If they are not meant to be used by other compilation units they should not be in the header file. The same as with `static` above: Proper scoping (i.e. using the most restrictive scope) is an important part of writing robust code.

Reply:
ok, I'll try to revert some of this.

I still see value in having a global 'header' file for general-purpose defines / macros that are more global in nature. Is this ok?

Reply:
As long as it's only used for defines, and does not include other headers this is fine.